lib/occi/api/client/http/authn_plugins/keystone.rb

```diff
@@ -74,9 +77,19 @@ def set_auth_token(tenant = nil)
             raise ::Occi::Api::Client::Errors::AuthnError,
                   "Unable to get Keystone API version from the response, fallback failed!" unless match && match[1]
             if match[1] == '2'
-              handler_class = KeystoneV2
+          if keystone_version == nil or keystone_version == 2
+                Occi::Api::Log.debug "Selecting Keystone V2 interface"
+                handler_class = KeystoneV2
+          else
```

Comment:
There is a tool called `rubocop` which will be used (and enforced) in the new version of all libraries. Here, however, it would be useless since you would drown in a sea of old transgressions :-)